File: leetcode-04/solutions/python3/0155.py
class MinStack:

    def __init__(self):
        self.stack = []

    def push(self, x: int) -> None:
        min_ = x if not self.stack else min(self.stack[-1][1], x)
        self.stack.append([x, min_])

    def pop(self) -> None:
        self.stack.pop()

    def top(self) -> int:
        return self.stack[-1][0]

    def getMin(self) -> int:
        return self.stack[-1][1]

File: leetcode-04/solutions/cpp/0155.cpp
class MinStack {
 public:
  void push(int x) {
    int min = stack.empty() ? x : std::min(stack.top().second, x);
    stack.push({x, min});
  }

  void pop() { stack.pop(); }

  int top() { return stack.top().first; }

  int getMin() { return stack.top().second; }

 private:
  std::stack<pair<int, int>> stack;
};